verilog/formatting/token_annotator.cc

```diff
@@ -849,10 +849,13 @@ static WithReason<SpacingOptions> BreakDecisionBetween(
   if (right.TokenEnum() == TK_else) {
     // TODO(fangism): feels like this should be the responsibility of
     // tree_unwrapper, handled by kElseClause, kGenerateElseClause, etc.
-    if (left.TokenEnum() == TK_end) {
+    if (left.TokenEnum() == TK_end && !style.wrap_end_else_clauses) {
       return {SpacingOptions::kMustAppend,
               "'end'-'else' and should be together on one line."};
     }
+    if (left.TokenEnum() == TK_end && style.wrap_end_else_clauses) {
```

Comment:
Test opportunity: if you change this line to
```
if (left.TokenEnum() == TK_end || style.wrap_end_else_clauses) 
```

... then no test actually notices that and triggers...